docs: Update shuffle sharding for ingest storage (#12480)
grafana/mimir: docs/sources/mimir/configure/configure-shuffle-sharding/index.md

Shuffle sharding assigns each tenant a shard that is composed of a subset of the Grafana Mimir instances or a subset of Kafka partitions.
This technique minimizes the number of overlapping instances between two tenants.
Shuffle sharding provides the following benefits:

- An outage on some Grafana Mimir cluster instances, Kafka partitions, or nodes only affects a subset of tenants.
- A misbehaving tenant only affects its shard instances or partitions.
  Assuming that each tenant shard is relatively small compared to the total number of instances or partitions in the cluster, it’s likely that any other tenant runs on different instances or partitions, or that only a subset of instances match the affected instances.

Using shuffle sharding doesn’t require more resources, but can result in unbalanced instances or partitions.

### Low overlapping instances or partition probability

For example, in a Grafana Mimir cluster that runs 50 Kafka partitions and assigns each tenant four out of 50 partitions, shuffling partitions between each tenant creates 230,000 possible combinations.

Randomly picking two tenants yields the following probabilities:

> - 71% chance that they don't share any partitions

Reviewer: Do we need to specify instances or partitions here so that we speak to both architectures?

{{< /admonition >}}

**Scaling up partitions:**
- New partitions start in PENDING state and are excluded from shuffle sharding
- Once partitions transition to ACTIVE state, they become eligible for shuffle sharding
- The ShuffleShardWithLookback algorithm ensures queries include both old and new partitions during the transition

**Scaling down partitions:**
- Partitions transition to INACTIVE state and stop receiving new writes
- INACTIVE partitions remain queryable during the lookback period
- After the configured time expires (`-ingester.partition-ring.delete-inactive-partition-after`, default 13 hours), each ingester removes itself and its partition from the ring automatically
- Note: This only removes the logical partition from the ring; the underlying Kafka partitions remain unchanged

{{< admonition type="note" >}}
Unlike the traditional ingester ring approach, partition shuffle sharding doesn't require zone-awareness since partition ownership is distributed across zones through the ingester instances.

> {{< /admonition >}}

Author: ```suggestion
```

- **Zone-awareness**<br />
  When you enable [zone-aware replication](../configure-zone-aware-replication/), the subset of instances selected for each tenant contains a balanced number of instances for each availability zone.

### Partitions shuffle sharding

Without shuffle sharding when using ingest storage, the Grafana Mimir distributor divides the received series among all active partitions. With shuffle-sharding the distributor divides each tenant's series among multiple partitions.

> Configuring shuffle sharding on the write path automatically also enables shuffle sharding on the read path too.

Reviewer: ```suggestion
Configuring shuffle sharding on the write path automatically enables it on the read path too.
```

> You can configure the default shard size for all tenants by setting - `-ingest-storage.ingestion-partition-tenant-shard-size=<size>` or its YAML equavalent. The shard size can be overridden on a per-tenant basis by setting `ingestion_partitions_tenant_shard_size` in the `overrides` section of the runtime configuration.

Reviewer: ```suggestion
You can configure the default shard size for all tenants by setting - `-ingest-storage.ingestion-partition-tenant-shard-size=<size>` or its YAML equavalent. You can override the shard size on a per-tenant basis by setting `ingestion_partitions_tenant_shard_size` in the `overrides` section of the runtime configuration.
```

#### Partitions shuffle sharding on the write path

> With partition shuffle sharding, the distributor divides each tenant's series among `-ingest-storage.ingestion-partition-tenant-shard-size` number of partitions.

Reviewer: ```suggestion
With partition shuffle sharding, the distributor divides each tenant's series among the number of partitions specified in `-ingest-storage.ingestion-partition-tenant-shard-size`.
```

#### Partitions shuffle sharding read path

> When shuffle sharding is enabled for the write path, the read path automatically uses the same configuration to query only the relevant partition owners (ingesters). The querier uses the ShuffleShardWithLookback algorithm to:

Reviewer: ```suggestion
When shuffle sharding is enabled for the write path, the read path automatically uses the same configuration to query only the relevant partition owners (ingesters). The querier uses the `ShuffleShardWithLookback` algorithm to:
```

- Include all partitions in the tenant's current shard
- Include partitions that were recently part of the tenant's shard (within the lookback period)
- Include INACTIVE partitions that are transitioning to a read-only state

> - Exclude PENDING partitions that have not yet received any traffic.

Reviewer: ```suggestion
- Exclude PENDING partitions that have not yet received any traffic
```

> This ensures query consistency during partition lifecycle transitions such as adding or removing ingesters. For more information about the hash ring and the state of partitions see [Hash ring](../../references/architecture/hash-ring/)

Reviewer: ```suggestion
This ensures query consistency during partition lifecycle transitions such as adding or removing ingesters. For more information about the hash ring and the state of partitions, refer to [Hash ring](../../references/architecture/hash-ring/).
```

#### Rollout strategy

> If you’re running a Grafana Mimir cluster with shuffle sharding disabled, and you want to enable it for the ingesters, use the following rollout strategy to avoid missing querying for any series currently in the ingesters:

Reviewer: ```suggestion
If you’re running a Grafana Mimir cluster with shuffle sharding disabled, and you want to enable it for the ingesters, use the following rollout strategy to avoid missing queries for any series currently in the ingesters:
```
